phpBB/includes/functions_download.php

```diff
@@ -41,15 +43,12 @@ function send_avatar_to_browser($file, $browser)
 	{
 		$image_dir = '';
 	}
-	$file_path = $phpbb_root_path . $image_dir . '/' . $prefix . $file;
+	$file_path = $image_dir . '/' . $prefix . $file;
 
-	if ((@file_exists($file_path) && @is_readable($file_path)) && !headers_sent())
+	if ($storage->exists($file_path) && !headers_sent())
 	{
 		header('Cache-Control: public');
 
-		$image_data = @getimagesize($file_path);
-		header('Content-Type: ' . image_type_to_mime_type($image_data[2]));
```

Comment:
https://flysystem.thephpleague.com/api/ (get mime types and get file sizes)
https://stackoverflow.com/questions/3910071/check-file-size-on-s3-without-downloading
for example